src/main/java/org/truffleruby/interop/OutgoingForeignCallNode.java

```diff
@@ -329,6 +331,22 @@ public Object executeCall(VirtualFrame frame, TruffleObject receiver, Object[] a
 
     }
 
+    protected class InspectOutgoingNode extends OutgoingNode {
+
+        @Child private StringNodes.MakeStringNode makeStringNode = StringNodes.MakeStringNode.create();
+
+        @Override
+        public Object executeCall(VirtualFrame frame, TruffleObject receiver, Object[] args) {
+            return makeStringNode.executeMake(inspect(receiver), UTF8Encoding.INSTANCE, CodeRange.CR_UNKNOWN);
+        }
+
+        @TruffleBoundary
+        private String inspect(TruffleObject receiver) {
+            return String.format("#<Truffle::Interop::Foreign:0x%x>", System.identityHashCode(receiver));
```

Comment:
I think the `0x` prefix is originally for memory addresses/object_id, I don't think it makes so much sense here (it could be confusing if two different objects have the same 0x... notation).
Maybe a Java notation like `#<Truffle::Interop::Foreign@hexhashcode>` would be good?
Otherwise maybe just removing the 0x prefix like mentioned in the .md is OK.